signer/src/transaction_coordinator.rs

```diff
@@ -1672,76 +1695,295 @@ where
-        tracing::info!("preparing pending requests for processing");
-
-        let storage = self.context.get_storage();
-        let config = self.context.config();
-
-        // Get the current sBTC limits (caps).
-        let sbtc_limits = self.context.state().get_current_limits();
+    pub async fn get_eligible_pending_withdrawal_requests<DB>(
+        storage: &DB,
+        expiry_window: u64,
+        soft_expiry_window: u64,
+        min_confirmations: u64,
+        params: &GetPendingRequestsParams<'_>,
+    ) -> Result<Vec<utxo::WithdrawalRequest>, Error>
+    where
+        DB: DbRead,
+    {
+        // Constants used for logging (local to this method).
+        const REQUEST_SKIPPED_MESSAGE: &str = "skipping withdrawal request";
+        const SKIP_REASON_AMOUNT_IS_DUST: &str = "amount_is_dust";
+        const SKIP_REASON_PER_WITHDRAWAL_CAP_EXCEEDED: &str = "per_withdrawal_cap_exceeded";
+        const SKIP_REASON_INSUFFICIENT_CONFIRMATIONS: &str = "insufficient_confirmations";
+        const SKIP_REASON_INSUFFICIENT_VOTES: &str = "insufficient_votes";
+        const SKIP_REASON_SOFT_EXPIRY: &str = "soft_expiry";
 
-        let context_window = self.context_window;
-        let accept_threshold = self.threshold;
-
-        // Fetch eligible deposit requests.
-        let pending_deposit_requests = storage
-            .get_pending_accepted_deposit_requests(
-                bitcoin_chain_tip.as_ref(),
-                context_window,
-                accept_threshold,
-            )
-            .await?;
+        let mut eligible_withdrawals = Vec::new();
 
         // Determine the minimum bitcoin block height we should consider for
         // withdrawals.
-        let min_bitcoin_height = bitcoin_chain_tip
+        let min_bitcoin_height = params
+            .bitcoin_chain_tip
             .block_height
-            .saturating_sub(WITHDRAWAL_BLOCKS_EXPIRY);
+            .saturating_sub(expiry_window);
 
-        // Fetch eligible withdrawal requests.
+        // We also calculate the minimum bitcoin block height for withdrawals
+        // that are considered valid (not expired) based on the soft expiry. We
+        // will not propose these withdrawals in the sweep transaction, but we
+        // will log them as skipped.
+        let min_soft_bitcoin_height = min_bitcoin_height.saturating_add(soft_expiry_window);
+
+        // Fetch pending withdrawal requests from storage. This method, with
+        // the given inputs, performs the following filtering according to
+        // consensus rules:
+        //
+        // - [1]  The request has not been swept in the canonical bitcoin chain,
+        // - [2]  Is confirmed in a canonical stacks block,
+        // - [4a] Is accepted by >= `threshold` signers (pre-filter),
+        // - [7]  Is not older than the provided withdrawal expiry threshold.
+        //
+        // We set the context window for the pending-accepted query below to the
```

Comment:
nit: we no longer deal with a context window below, but it could be intended as the one induced by `min_bitcoin_height`, so it could stay as is. Feel free to resolve in that case.

Reply:
Nah it's better it's correct, reworded in [6ff4487](https://github.com/stacks-network/sbtc/pull/1399/commits/6ff4487930b3a796612ec7f08045ed1def810a2b)